File: android/expoview/src/main/java/abi27_0_0/host/exp/exponent/modules/api/components/gesturehandler/react/RNGestureHandlerEventDataExtractor.java
package abi27_0_0.host.exp.exponent.modules.api.components.gesturehandler.react;

import abi27_0_0.com.facebook.react.bridge.WritableMap;
import abi27_0_0.host.exp.exponent.modules.api.components.gesturehandler.GestureHandler;

public interface RNGestureHandlerEventDataExtractor<T extends GestureHandler> {
  void extractEventData(T handler, WritableMap eventData);
}

File: android/vendored/sdk49/react-native-gesture-handler/android/src/main/java/abi49_0_0/com/swmansion/gesturehandler/react/RNGestureHandlerEventDataExtractor.kt
package abi49_0_0.com.swmansion.gesturehandler.react

import abi49_0_0.com.facebook.react.bridge.WritableMap
import abi49_0_0.com.swmansion.gesturehandler.core.GestureHandler

interface RNGestureHandlerEventDataExtractor<T : GestureHandler<T>> {
  fun extractEventData(handler: T, eventData: WritableMap)
}
